### PR TITLE
Authorize Mismatch Finder to identify users against Wikidata

### PR DESCRIPTION
**Pull request by itamargiv**
This PR Attempts to enable the authorization of users against wikidata, to couple user details with registered wikidata users and rely on Wikimedia logins instead of creating our own password login system. 

Bug: [T285327](https://phabricator.wikimedia.org/T285327)

### CODE REVIEW

**Review by Silvan-WMDE — DISMISSED** [review]
Looks good to me and works locally. Two minor comments below.

File: docs/diagrams/mismatch_application.drawio.svg [hunk 1]
```diff
@@ -224,61 +224,62 @@
...
+        <path d="M 611.88 485 L 604.88 488.5 L 606.63 485 L 604.88 481.5 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/>
+        <path d="M 584.12 485 L 591.12 481.5 L 589.37 485 L 591.12 488.5 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/>
+        <path d="M 613 443 L 613 420 L 763 420 L 763 443" fill="#ffffff" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/>
+        <path d="M 613 443 L 613 550 L 763 550 L 763 443" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/>
+        <path d="M 613 443 L 763 443" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/>
         <g fill="#000000" font-family="Helvetica" font-weight="bold" pointer-events="none" text-anchor="middle" font-size="18px">
-            <text x="502.5" y="459">
-                User Provider
+            <text x="687.5" y="439">
+                User
```
[the remark this answers is not in the document]
Comment (itamargiv) [reply docs/diagrams/mismatch_application.drawio.svg:263]: Good point, thank you!

File: routes/web.php [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <?php
 
 use Illuminate\Support\Facades\Route;
+use App\Http\Controllers\OAuthUserController;
```
[the remark this answers is not in the document]
Comment (itamargiv) [reply routes/web.php:4]: indeed, we should create a task to apply the linter

[the remark this answers is not in the document]
Comment (itamargiv) [reply routes/web.php:4]: [T285746](https://phabricator.wikimedia.org/T285746) Created for linitng

[the remark this answers is not in the document]
Comment (Silvan-WMDE) [reply routes/web.php]: Great, thanks for the ticket!

**Pull request merged**